driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```
@@ -0,0 +1,164 @@
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, Boolean>() {
+            @Override
+            public Boolean apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return queryBatchCursor.hasNext();
+            }
+        });
+    }
+
+    @Override
+    public List<T> next() {
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.next());
+            }
+        });
+    }
+
+    @Override
+    public List<T> tryNext() {
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.tryNext());
+            }
+        });
+    }
+
+    @Override
+    public void close() {
+        wrapped.close();
+    }
+
+    @Override
+    public void setBatchSize(final int batchSize) {
+        wrapped.setBatchSize(batchSize);
+    }
+
+    @Override
+    public int getBatchSize() {
+        return wrapped.getBatchSize();
+    }
+
+    @Override
+    public ServerCursor getServerCursor() {
+        return wrapped.getServerCursor();
+    }
+
+    @Override
+    public ServerAddress getServerAddress() {
+        return wrapped.getServerAddress();
+    }
+
+    @Override
+    public void remove() {
+        throw new UnsupportedOperationException("Not implemented yet!");
+    }
+
+    private List<T> convertResults(final List<RawBsonDocument> rawDocuments) {
+        List<T> results = null;
+        if (rawDocuments != null) {
+            results = new ArrayList<T>();
+            for (RawBsonDocument rawDocument : rawDocuments) {
+                resumeToken = rawDocument.getDocument("_id");
```

The spec says "A driver MUST raise an error on the first response received without a resume token".  I read that as saying that this should throw a MongoChangeStreamException if _id is missing or is not a document.  As is, it will throw a BsonInvalidOperationException. 

And doing that here means you can get rid of the check for a missing resumeToken that'
s now in ChangeStreamOperation.